.travis.yml

```diff
@@ -0,0 +1,114 @@
+# Abstracted from uBoot travis.yml (Why reinvent the wheel?)
+
+language: c
+sudo: required
+dist: trusty
+
+addons:
+  apt:
+    sources:
+    - ubuntu-toolchain-r-test
+    - llvm-toolchain-trusty-7
+    packages:
+    - bison
+    - flex
+    - cppcheck
+    - sparse
+    - bc
+    - swig
+    - device-tree-compiler
+    - lzop
+    - liblz4-tool
+    - lzma-alone
+    - libisl15
+    - libelf-dev
+
+install:
+  - sudo apt-get update -qq
+
+env:
+  global:
+    - export BUILD_THREADS=$(grep "^processor" /proc/cpuinfo | wc -l)
+    - export EXTRA_FLAGS="CONFIG_COMPLIE_TEST=y"
+
+before_script:
+  - if [[ "${TRAVIS_BRANCH}" != ${TARGET_BRANCH} ]]; then exit 0 ; fi
+  - mkdir -p toolchain
+  - cd toolchain
+  - echo ${EXTRA_FLAGS}
+  - if [[ "${TOOLCHAIN}" == arm ]]; then
+     wget https://developer.arm.com/-/media/Files/downloads/gnu-a/8.3-2019.03/binrel/gcc-arm-8.3-2019.03-x86_64-arm-linux-gnueabihf.tar.xz &&
+     tar xf gcc-arm-8.3-2019.03-x86_64-arm-linux-gnueabihf.tar.xz;
+   fi
+  - if [[ "${TOOLCHAIN}" == arm64 ]]; then
+      wget https://developer.arm.com/-/media/Files/downloads/gnu-a/8.3-2019.03/binrel/gcc-arm-8.3-2019.03-x86_64-aarch64-linux-gnu.tar.xz &&
+      tar xf gcc-arm-8.3-2019.03-x86_64-aarch64-linux-gnu.tar.xz;
+    fi
+  - if [[ "${TOOLCHAIN}" == x86_64 ]]; then
+      wget https://mirrors.edge.kernel.org/pub/tools/crosstool/files/bin/x86_64/8.1.0/x86_64-gcc-8.1.0-nolibc-x86_64-linux.tar.gz &&
+      tar xf x86_64-gcc-8.1.0-nolibc-x86_64-linux.tar.gz;
+    fi
+  - cd ..
+
+script:
+  - if [[ "${TOOLCHAIN}" == arm64 ]]; then
+     export ARCH=arm64;
+     export CROSS_COMPILE=$PWD/toolchain/gcc-arm-8.3-2019.03-x86_64-aarch64-linux-gnu/bin/aarch64-linux-gnu-;
+     make -j${BUILD_THREADS} defconfig;
+     make -j${BUILD_THREADS} ${EXTRA_FLAGS} ${TARGET_MODULE};
+    fi
+  - if [[ "${TOOLCHAIN}" == arm ]]; then
+     export ARCH=arm;
+     export CROSS_COMPILE=$PWD/toolchain/gcc-arm-8.3-2019.03-x86_64-arm-linux-gnueabihf/bin/arm-linux-gnueabihf-;
+     make -j${BUILD_THREADS} multi_v7_defconfig;
+     make -j${BUILD_THREADS} ${EXTRA_FLAGS};
+    fi
+  - if [[ "${TOOLCHAIN}" == x86_64 ]]; then
+     export CROSS_COMPILE=$PWD/toolchain/gcc-8.1.0-nolibc/x86_64-linux/bin/x86_64-linux-;
+     make -j${BUILD_THREADS} x86_64_defconfig;
+     make -j${BUILD_THREADS} ${EXTRA_FLAGS};
+    fi
+
+matrix:
+  include:
+  # we need to build by vendor due to 50min time limit for builds
+  # each env setting here is a dedicated build
+    - name: "arm full build"
+      env:
+        - TOOLCHAIN="arm"
+          TARGET_BRANCH="master"
+    - name: "arm64 full build"
+      env:
+        - TOOLCHAIN="arm64"
+          TARGET_BRANCH="master"
+    - name: "x86_64 full build"
+      env:
+        - TOOLCHAIN="x86_64"
+          TARGET_BRANCH="master"
+    - name: "drm bridge partial"
+      env:
+        - TOOLCHAIN="arm64"
+          TARGET_BRANCH="travis_baseline"
+          TARGET_MODULE="drivers/gpu/drm/bridge/"
+    - name: "phy ti partial"
```

Comment:
Could we make the TARGET_MODULE as an optional parameter than can be defined by developer?

Reply:
TARGET_MODULE is an optional parameter per our documentation (see below)
No there is no way to inject command line arguments from the developer.  Whatever is in the yaml file is what travis will execute.  The yaml is non-executable and does not take arguments.  So if we want more modules we need to add them.  

yml Matrix Build Tag
matrix:

   include:
       - name: "arm full build"
         env:
              - TOOLCHAIN="arm"
                TARGET_BRANCH="master"

       - name: "phy ti partial"
         env:
              - TOOLCHAIN="arm64"
              TARGET_BRANCH="travis_baseline"
              TARGET_MODULE="drivers/phy/ti/"
              EXTRA_FLAGS="C=2 "${EXTRA_FLAGS}

matrix and include are documented here https://docs.travis-ci.com/user/build-matrix/.

name is the name of the build that will appear in the travis-ci.org dashboard.

env defines the environment variables for this specific build.  Following this label are TI defined environment variables.

TI Defined Required Variables
TOOLCHAIN - Defines the build type as either "arm", "arm64" or "x86_64"

TARGET_BRANCH - Defines the branch that this specific test case needs to be run under.

TI Defined Optional Variables
TARGET_MODULES - This defines the specific modules to compile. 

For example: If only the USB drivers should be compiled then this variable will be TARGET_MODULE="drivers/usb/"

If a full build of the kernel is required then this variable should not be defined.

EXTRA_FLAGS - This define any addition make parameters that the build should use.  The TI yml file defaults EXTRA_FLAGS to "CONFIG_COMPLIE_TEST=y"

If this compile test is not needed then the EXTRA_FLAGS can be cleared by setting EXTRA_FLAGS=""